multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/mmrcms/MMRCMS.kt

```diff
@@ -1,529 +1,474 @@
+ */
+abstract class MMRCMS @Suppress("UNUSED") constructor(
     override val name: String,
     override val baseUrl: String,
-    override val lang: String,
-    sourceInfo: String = "",
-) : HttpSource() {
-    open val jsonData = if (sourceInfo == "") {
-        SourceData.giveMetaData(baseUrl)
-    } else {
-        sourceInfo
-    }
+    final override val lang: String,
 
-    /**
-     * Parse a List of JSON sources into a list of `MyMangaReaderCMSSource`s
-     *
-     * Example JSON :
-     * ```
-     *     {
-     *         "language": "en",
-     *         "name": "Example manga reader",
-     *         "base_url": "https://example.com",
-     *         "supports_latest": true,
-     *         "item_url": "https://example.com/manga/",
-     *         "categories": [
-     *             {"id": "stuff", "name": "Stuff"},
-     *             {"id": "test", "name": "Test"}
-     *         ],
-     *         "tags": [
-     *             {"id": "action", "name": "Action"},
-     *             {"id": "adventure", "name": "Adventure"}
-     *         ]
-     *     }
-     *
-     *
-     * Sources that do not supports tags may use `null` instead of a list of json objects
-     *
-     * @param sourceString The List of JSON strings 1 entry = one source
-     * @return The list of parsed sources
-     *
-     * isNSFW, language, name and base_url are no longer needed as that is handled by multisrc
-     * supports_latest, item_url, categories and tags are still needed
-     *
-     *
-     */
-    private val json: Json by injectLazy()
-    val jsonObject = json.decodeFromString<JsonObject>(jsonData)
-    override val supportsLatest = jsonObject["supports_latest"]!!.jsonPrimitive.boolean
-    open val itemUrl = jsonObject["item_url"]!!.jsonPrimitive.content
-    open val categoryMappings = mapToPairs(jsonObject["categories"]!!.jsonArray)
-    open var tagMappings = jsonObject["tags"]?.jsonArray?.let { mapToPairs(it) } ?: emptyList()
+    vararg forbiddens: Forbidden,
 
-    /**
-     * Map an array of JSON objects to pairs. Each JSON object must have
-     * the following properties:
-     *
-     * id: first item in pair
-     * name: second item in pair
-     *
-     * @param array The array to process
-     * @return The new list of pairs
-     */
-    open fun mapToPairs(array: JsonArray): List<Pair<String, String>> = array.map {
-        it as JsonObject
+    protected val dateFormat: SimpleDateFormat = SimpleDateFormat("d MMM. yyyy", Locale.US),
+    protected val itemPath: String = "manga",
+    protected val fetchFilterOptions: Boolean = true,
+    protected val supportsAdvancedSearch: Boolean = true,
+    protected val detailsTitleSelector: String = ".listmanga-header, .widget-title",
+    protected val chapterNamePrefix: String = "",
+    protected val chapterString: String = when (lang) {
+        "es" -> "Capítulo"
+        "fr" -> "Chapitre"
+        else -> "Chapter"
+    },
```

Comment:
Make them `private`, otherwise it defeats the purpose of moving them into the constructor. Subclasses should definitely have no need to override them anymore.

Reply:
Keeping `itemPath` `protected` for Mangas.in.